### PR TITLE
BUG: corner cases in DTI.get_value, Float64Index.get_value

### PR DESCRIPTION
**Pull request by jbrockmendel**
Series lookups are affected for the Float64Index case.

### CODE REVIEW

File: pandas/tests/indexes/datetimes/test_indexing.py [hunk 1]
```diff
@@ -773,3 +773,14 @@ def test_get_loc_nat(self):
         # GH#20464
         index = DatetimeIndex(["1/3/2000", "NaT"])
         assert index.get_loc(pd.NaT) == 1
+
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_indexing.py]: maybe better to parmateize over the accepted nulls (do we have a fixture for date time nulls?) and then have an errors version

Comment (jbrockmendel) [reply pandas/tests/indexes/datetimes/test_indexing.py]: ill put that on my list of follow-ups, we're about 2/3 of the way towards having get_value be internally consistent.

Thoughts on the Series.at outlier?

Comment (jreback) [reply pandas/tests/indexes/datetimes/test_indexing.py]: what’s the Series.at outlier?

File: pandas/tests/indexes/test_numeric.py [hunk 2]
```diff
@@ -393,6 +393,46 @@ def test_get_loc_missing_nan(self):
...
+        [
+            pd.date_range("2016-01-01", periods=3),
+            pd.timedelta_range("1 Day", periods=3),
+        ],
+    )
+    def test_lookups_datetimelike_values(self, vals):
+        # If we have datetime64 or timedelta64 values, make sure they are
+        #  wrappped correctly
+        ser = pd.Series(vals, index=range(3, 6))
+        ser.index = ser.index.astype("float64")
```
Comment (MomIsBestFriend) [inline pandas/tests/indexes/test_numeric.py]: Will this work on 32-bit version as well? (not a blocker or anything like that, just asking for getting more knowledge)

Comment (jreback) [reply pandas/tests/indexes/test_numeric.py]: we don't support non-64bit Indexes currently, so yes.

File: pandas/tests/indexes/datetimes/test_indexing.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/indexes/datetimes/test_indexing.py]: > ill put that on my list of follow-ups, we're about 2/3 of the way towards having get_value be internally consistent.
> 
sure

**Review by jreback — COMMENTED** [review]
looks fine.

File: pandas/tests/indexes/test_numeric.py (hunk 2 above)
Comment (MomIsBestFriend) [reply pandas/tests/indexes/test_numeric.py]: Thank you for the explanation @jreback

File: pandas/tests/indexes/datetimes/test_indexing.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/indexes/datetimes/test_indexing.py]: > what’s the Series.at outlier?

in the test here `ser.loc[4]` behaves like `ser.loc[4.0]`, while `ser.at[4]` raises

File: pandas/tests/indexes/test_numeric.py [hunk 3]
```diff
@@ -393,6 +393,46 @@ def test_get_loc_missing_nan(self):
...
+        assert isinstance(result, type(expected)) and result == expected
+
+        result = ser.loc[4.0]
+        assert isinstance(result, type(expected)) and result == expected
+        result = ser.loc[4]
+        assert isinstance(result, type(expected)) and result == expected
+
+        result = ser.at[4.0]
+        assert isinstance(result, type(expected)) and result == expected
+        # Note: ser.at[4] raises ValueError; TODO: should we make this match loc?
```
Comment (jreback) [inline pandas/tests/indexes/test_numeric.py]: yes this should match .loc

Comment (jbrockmendel) [reply pandas/tests/indexes/test_numeric.py]: sounds good, should be a small patch, so quick turnaround

**Comment by jbrockmendel** [comment]
gentle ping; this a blocker before fixing the .at bug

File: pandas/tests/indexes/test_numeric.py (hunk 3 above)
Comment (jreback) [reply pandas/tests/indexes/test_numeric.py]: this PR looks good, is this a followup (the .at issue)?

Comment (jbrockmendel) [reply pandas/tests/indexes/test_numeric.py]: exactly

**Pull request merged**